test/unit/PeerSelectControllerSpec.js

```diff
@@ -0,0 +1,258 @@
+describe('PeerSelectController', function () {
+  var $controller, $scope, $q, $mod, $APManager, $EService;
+  var createController, timeoutTime;
+  var $promiseData, $promise, $promiseFlag;
+
+  beforeEach(module('myApp.controllers'));
+
+  beforeEach(function () {
+    // The modalInstance will propably usually give a boolean as return.
+    // However, for testing purposes it is important to gain knowledge about the input of the function
+    $mod = {
+      close: function (arr) {
+        return arr;
+      }
+    };
+
+    timeoutTime = 1000;
+
+    $APManager = {
+      getPeerString: function (str){
+        return "P".concat(str);
+      },
+      getPeerID: function(str) {
+        return str.slice(-1);
+      },
+      getPeer: function (id) {
+        return id.concat("peer");
+      }
+    };
+
+    // The controller is created in the test in order to test different initial content of scope variables.
+    createController = function() {
+      $controller('PeerSelectController', {
+        $scope: $scope,
+        $modalInstance: $mod,
+        $q: $q,
+        AppPeersManager: $APManager,
+        ErrorService: $EService
+      });
+    };
+
+    $promiseFlag = false;
+    $promise = {
+      then: function (f) {
+        $promiseFlag = true;
+        f();
+      }
+    };
+
+    $EService = {
+      confirm: function(data){
+        $promiseData = data;
+        return $promise;
+      }
+    };
+
+    $q = {
+      when: function () {
+        return $promise;
+      }
+    };
+
+    inject(function (_$controller_, _$rootScope_) {
+      $controller = _$controller_;
+      $scope = _$rootScope_.$new();
+    });
+  });
+
+  it('compiles', function (done) {
+    createController();
+
+    // Set timer to give the controller time to resolve.
+    setTimeout(function(){
+      expect($scope.selectedPeers).toBeDefined();
+      expect($scope.selectedPeersIDs).toBeDefined();
+      expect($scope.selectedCount).toBeDefined();
+    }, timeoutTime);
+
+    done();
+  });
+
+  it('compiles with a shareLinkPromise that resolves', function (done) {
+    $scope.shareLinkPromise = {
+      then: function (resolve, reject) {
+        setTimeout(resolve("test"), timeoutTime);
```

Comment:
I feel like you should move that string into a variable. I was surprised to find an 'url' "test" below.